### ISSUE TITLE
I keep getting a InvalidAuthenticityToken error. Could it be because of Rails 5.2.0?

### ISSUE THREAD

**Issue by jakeyeaton16** [Shopify/shopify_app#553]
Rails **5.2.0**
Ruby: **2.4.1**
Browser: **Chrome**
Operating System: **Mac**
Hosting Provider: **DigitalOcean**

I'm trying to create a test Shopify app.

But I keep running into the following error in my logs: 

1. I followed the "Build a Rails App With the Shopify Gem in Five Minutes | Shopify Partners" guide
2. Uploaded my code to GitHub then Hatchbox and using DigitalOcean
3. When I go to my given URL I get an error page. So when I look at the rails logs on Hatchbox the four lines which say 'failed': 

`F, [2018-03-05T23:34:56.533678 #23942] FATAL -- : [a7f46489-9098-48de-a2c3-16325b81744d]   
F, [2018-03-05T23:34:56.533738 #23942] FATAL -- : [a7f46489-9098-48de-a2c3-16325b81744d] ActionController::InvalidAuthenticityToken (ActionController::InvalidAuthenticityToken):
F, [2018-03-05T23:34:56.533795 #23942] FATAL -- : [a7f46489-9098-48de-a2c3-16325b81744d]   
F, [2018-03-05T23:34:56.533877 #23942] FATAL -- : [a7f46489-9098-48de-a2c3-16325b81744d] actionpack (5.2.0.rc1) lib/action_controller/metal/request_forgery_protection.rb:211:in `handle_unverified_request'`

I'm not sure what is wrong. Could this be because I'm using Rails 5.2.0? 

If you need more information feel free to ask. My repo is private and has my secret API key, thus why I haven't shared it on this issue yet.

**Comment by vfonic** [comment]
I'd suggest you try with rails 5.1. It could be due to rails 5.2 easily.

I guess you know this, but I had it at hand:
```sh
rails _5.1.5_ new app_testing --database=postgresql
cd app_testing
echo 'gem "shopify_app"' >> Gemfile
bundle install
rails generate shopify_app --api_key <your_api_key> --secret <your_app_secret>
```

**Comment by jakeyeaton16** [comment]
Yes, that fixed the issue!!!

Thank you so much 👍

**Comment by vfonic** [comment]
No problem! :)

I hope the issue is fixed soon so that the gem works with rails 5.2.x.

**Comment by jakeyeaton16** [comment]
Yes me too!

**Comment by 0ptimusrhyme** [comment]
I'm pretty sure this is due to the fact that we don't use rails form helpers when constructing the sessions#new view. I've been able to fix this locally by replacing the HTML form with the helper methods:

```erb
<%= form_tag url: {action: "create"} do %>
      <% if flash[:error] %>
        <div class=error><%= flash[:error] %></div>
      <% end %>
      <%= text_field_tag :shop, nil, { placeholder: "example.myshopify.com", id: "shop", autofocus: "autofocus", class: "marketing-input" } %>
      <%= button_tag "Install", { class: "marketing-button" } %>
<% end %>
```

The exception was being thrown because [this method](https://github.com/rails/rails/blob/89bcca59e91fa9da941de890012872e8288e77b0/actionpack/lib/action_controller/metal/request_forgery_protection.rb#L288) was evaluating to `false`, because no authenticity token was being set on the form.

Accessing Sessions#new with the URL param `?shop=shop.myshopify.com` would not cause the error which led me to investigating the form elements themselves.

I'm going to throw together a PR to address this and will link it here.

**Comment by 0ptimusrhyme** [comment]
Are you running version 8.2.5 of the gem? I unwittingly was troubleshooting this on 8.2.4 and I even came to the same conclusion found in #534. Facepalm.

This was addressed in version 8.2.5, the problem was that an authenticity token wasn't being created for the form in the Sessions#new view. This was because we weren't using Rails form helpers to build the form, causing a failure when ActionController tries to check the authenticity token provided by the form (`nil` in this case).

Make sure the gem is up to date! Feel free to reopen if you feel like I've missed something.

**Comment by benbonnet** [comment]
@0ptimusrhyme  8.2.6 and still getting the issue :/
